js/src/edit-block/components/fields-grid.js

```diff
@@ -0,0 +1,187 @@
+				newFieldNumber.toString()
+			)
+			: __( 'New Field', 'genesis-custom-blocks' );
+
+		const newField = {
+			name,
+			label,
+			control: 'text',
+			type: 'string',
+			order: Object.values( fields ).length,
+		};
+
+		fields[ name ] = newField;
+		changeBlock( 'fields', fields );
+	}, [ block, changeBlock ] );
+
+	/**
+	 * Gets the fields for either the editor or inspector.
+	 *
+	 * @return {Array} The fields with the given location.
+	 */
+	const getFieldsForLocation = useCallback( () => {
+		if ( ! block || ! block.fields ) {
+			return null;
+		}
+
+		return getFieldsAsArray( block.fields ).filter( ( field ) => {
+			if ( isEditorDisplay ) {
+				return ! field.location || 'editor' === field.location;
+			}
+
+			return field.location === 'inspector';
+		} );
+	}, [ block, isEditorDisplay ] );
+
+	const fields = getFieldsForLocation();
+	const buttonClass = 'focus:outline-none h-12 px-4 text-sm';
+
+	return (
+		<>
+			<div className="flex mt-6">
+				<button
+					className={ buttonClass }
+					onClick={ () => {
+						setIsEditorDisplay( true );
+					} }
+				>
+					<span
+						className={ className( {
+							'font-semibold': isEditorDisplay,
+						} ) }
+					>
+						{ __( 'Editor Fields', 'genesis-custom-blocks' ) }
+					</span>
+				</button>
+				<button
+					className={ buttonClass }
+					onClick={ () => {
+						setIsEditorDisplay( false );
+					} }
+				>
+					<span
+						className={ className( {
+							'font-semibold': ! isEditorDisplay,
+						} ) }
+					>
+						{ __( 'Inspector Fields', 'genesis-custom-blocks' ) }
+					</span>
+				</button>
+			</div>
+			<div
+				role="grid"
+				className="grid grid-cols-4 gap-4 w-full items-start mt-2"
+			>
+				{
+					fields && fields.length
+						? fields.map( ( field, index ) => {
+							const selectField = () => {
+								setSelectedFieldName( field.name );
+							};
+
+							return (
+								<div
+									className={ className(
+										'field w-full',
+										getWidthClass( field.width )
+									) }
+									key={ `field-item-${ index }` }
+									role="gridcell"
+									tabIndex="0"
+									aria-label={
+										sprintf(
+											// translators: %s: the label of the field
+											__( 'Field: %s', 'genesis-custom-blocks' ),
+											field.label
+										)
+									}
+									onClick={ selectField }
+									onKeyPress={ selectField }
+								>
+									<div
+										className="relative flex items-center w-full p-4 bg-white border border-gray-400 rounded-sm hover:border-black"
+										id={ `field-item-${ index }` }
+									>
+										<button className="field-resize w-4 absolute -ml-2 left-0 top-0 bottom-0 focus:outline-none"></button>
+										<button>
+											<svg className="fill-current h-6 w-6" viewBox="0 0 24 24"><path d="M0 0h24v24H0z" fill="none" /><path d="M5 17v2h14v-2H5zm4.5-4.2h5l.9 2.2h2.1L12.75 4h-1.5L6.5 15h2.1l.9-2.2zM12 5.98L13.87 11h-3.74L12 5.98z" /></svg>
```

Comment:
If it's alright, we'll add different icons for each field in a later PR. Right now, all field types have the same:

![same-icon](https://user-images.githubusercontent.com/4063887/99357720-03cbc200-2872-11eb-8061-da50f347e196.png)